<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from artist.xacro                   | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="artist" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!--END OF MACROS-->
  <link name="base_link">
    <inertial>
      <mass value="1024"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="170.667" ixy="0" ixz="0" iyy="170.667" iyz="0" izz="170.667"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.1 0.5 0.5"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.1 0.5 0.5"/>
      </geometry>
    </visual>
  </link>
  <joint name="base_link__link_00" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-0.78" upper="0.78" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="link_00"/>
  </joint>
  <transmission name="trans_base_link__link_00">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="base_link__link_00">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_base_link__link_00">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="link_00">
    <inertial>
      <mass value="157.633"/>
      <origin rpy=" 0 0 0" xyz="0 0 0"/>
      <inertia ixx="13.235" ixy="0" ixz="0" iyy="13.235" iyz="0" izz="9.655"/>
    </inertial>
    <collision>
      <origin rpy=" 0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.1" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy=" 0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.1" radius="0.05"/>
      </geometry>
    </visual>
  </link>
  <joint name="link_00__link_01" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-1.57079" upper="1.57079" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link_00"/>
    <child link="link_01"/>
  </joint>
  <transmission name="trans_link_00__link_01">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="link_00__link_01">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_link_00__link_01">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="link_01">
    <inertial>
      <mass value="157.633"/>
      <origin rpy=" 1.57079 0 0" xyz="0 0 0"/>
      <inertia ixx="13.235" ixy="0" ixz="0" iyy="13.235" iyz="0" izz="9.655"/>
    </inertial>
    <collision>
      <origin rpy=" 1.57079 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.1" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy=" 1.57079 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.1" radius="0.05"/>
      </geometry>
    </visual>
  </link>
  <joint name="link_01__link_02" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-0.74" upper="0.74" velocity="0.5"/>
    <origin rpy="0 0 -0.7853" xyz="0 0 0"/>
    <parent link="link_01"/>
    <child link="link_02"/>
  </joint>
  <transmission name="trans_link_01__link_02">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="link_01__link_02">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_link_01__link_02">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="link_02">
    <inertial>
      <mass value="157.633"/>
      <origin rpy="0 -1.57079 0" xyz="0.25 0 0"/>
      <inertia ixx="13.235" ixy="0" ixz="0" iyy="13.235" iyz="0" izz="9.655"/>
    </inertial>
    <collision>
      <origin rpy="0 -1.57079 0" xyz="0.25 0 0"/>
      <geometry>
        <cylinder length="0.5" radius="0.02"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 -1.57079 0" xyz="0.25 0 0"/>
      <geometry>
        <cylinder length="0.5" radius="0.02"/>
      </geometry>
    </visual>
  </link>
  <joint name="link_01__link_03" type="revolute">
    <parent link="link_01"/>
    <child link="link_03"/>
    <origin rpy="0 0 -0.7853" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="0.5293" lower="-1.57" upper="1.57" velocity="0.5"/>
    <mimic joint="link_01__link_02" multiplier="-1" offset="0"/>
  </joint>
  <link name="link_03">
    <inertial>
      <mass value="57.906"/>
      <origin rpy="1.57079 0 0" xyz="0 0.25 0"/>
      <inertia ixx="12.679" ixy="0" ixz="0" iyy="12.679" iyz="0" izz="0.651"/>
    </inertial>
    <collision>
      <origin rpy="1.57079 0 0" xyz="0 0.25 0"/>
      <geometry>
        <cylinder length="0.5" radius="0.02"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.57079 0 0" xyz="0 0.25 0"/>
      <geometry>
        <cylinder length="0.5" radius="0.02"/>
      </geometry>
    </visual>
  </link>
  <joint name="link_03__link_04" type="revolute">
    <parent link="link_03"/>
    <child link="link_04"/>
    <origin rpy="0 0 0" xyz="0 0.5 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="0.5293" lower="-1.57" upper="1.57" velocity="0.5"/>
    <mimic joint="link_01__link_02" multiplier="2" offset="0"/>
  </joint>
  <link name="link_04">
    <inertial>
      <mass value="57.906"/>
      <origin rpy="0 -1.57079 0" xyz="0.25 0 0"/>
      <inertia ixx="12.679" ixy="0" ixz="0" iyy="12.679" iyz="0" izz="0.651"/>
    </inertial>
    <collision>
      <origin rpy="0 -1.57079 0" xyz="0.25 0 0"/>
      <geometry>
        <cylinder length="0.5" radius="0.02"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 -1.57079 0" xyz="0.25 0 0"/>
      <geometry>
        <cylinder length="0.5" radius="0.02"/>
      </geometry>
    </visual>
  </link>
  <joint name="link_02__link_05" type="revolute">
    <parent link="link_02"/>
    <child link="link_05"/>
    <origin rpy="0 0 0" xyz="0.5 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="0.5293" lower="-1.57" upper="1.57" velocity="0.5"/>
    <mimic joint="link_01__link_02" multiplier="-2" offset="0"/>
  </joint>
  <link name="link_05">
    <inertial>
      <mass value="57.906"/>
      <origin rpy="1.57 0 0" xyz="0 0.25 0"/>
      <inertia ixx="12.679" ixy="0" ixz="0" iyy="12.679" iyz="0" izz="0.651"/>
    </inertial>
    <collision>
      <origin rpy="1.57 0 0" xyz="0 0.25 0"/>
      <geometry>
        <cylinder length="0.5" radius="0.02"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.57 0 0" xyz="0 0.25 0"/>
      <geometry>
        <cylinder length="0.5" radius="0.02"/>
      </geometry>
    </visual>
  </link>
  <joint name="link_05__link_06" type="revolute">
    <parent link="link_05"/>
    <child link="link_06"/>
    <origin rpy="0 0 -0.78" xyz="0 0.5 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="0.5293" lower="-1.57" upper="1.57" velocity="0.5"/>
    <mimic joint="link_01__link_02" multiplier="1" offset="0"/>
  </joint>
  <link name="link_06">
    <inertial>
      <mass value="57.906"/>
      <origin rpy="1.57 0 0" xyz="0 0.125 0"/>
      <inertia ixx="12.679" ixy="0" ixz="0" iyy="12.679" iyz="0" izz="0.651"/>
    </inertial>
    <collision>
      <origin rpy="1.57 0 0" xyz="0 0.125 0"/>
      <geometry>
        <cylinder length="0.25" radius="0.02"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.57 0 0" xyz="0 0.125 0"/>
      <geometry>
        <cylinder length="0.25" radius="0.02"/>
      </geometry>
    </visual>
  </link>
  <joint name="link_06__link_07" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-3.14" upper="3.14" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link_06"/>
    <child link="link_07"/>
  </joint>
  <transmission name="trans_link_06__link_07">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="link_06__link_07">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_link_06__link_07">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="link_07">
    <inertial>
      <mass value="57.906"/>
      <origin rpy="0 0 0" xyz="0 0.25 0"/>
      <inertia ixx="12.679" ixy="0" ixz="0" iyy="12.679" iyz="0" izz="0.651"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.25 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0.25 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
    </visual>
  </link>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      </plugin>
  </gazebo>
</robot>
